python/grass/jupyter/timeseriesmap.py

```diff
@@ -410,25 +406,26 @@ def animate(
         formats, visit:
         https://pillow.readthedocs.io/en/stable/reference/ImageColor.html#color-names
 
+        param str filename: name of output GIF file
         param int duration: time to display each frame; milliseconds
         param bool label: include date/time stamp on each frame
         param str font: font file
         param int text_size: size of date/time text
         param str text_color: color to use for the text.
-        param str filename: name of output GIF file
         """
         # Create a GIF from the PNG images
         import PIL.Image  # pylint: disable=import-outside-toplevel
         import PIL.ImageDraw  # pylint: disable=import-outside-toplevel
         import PIL.ImageFont  # pylint: disable=import-outside-toplevel
-        import IPython.display  # pylint: disable=import-outside-toplevel
 
         # Render images if they have not been already
         if not self._layers_rendered:
             self.render()
 
         # filepath to output GIF
-        if not filename:
+        if filename:
+            assert filename.endswith(".gif"), "filename must end in '.gif'"
```

Comment:
This is possible, but asserts are usually not used for input checking in Python because they can be optimized out.